### PR TITLE
Lp port mp sale floor price

### PR DESCRIPTION
**Pull request by vrenaville**
Port of the lp branch 

lp:~camptocamp/sale-financial/7.0-port-sale_floor_price

### CODE REVIEW

**Comment by eLBati** [comment]
Hello @vrenaville some PEP8 remarks:

```
./sale_floor_price/__openerp__.py:21:8: E203 whitespace before ':'

./sale_floor_price/__openerp__.py:22:11: E203 whitespace before ':'

./sale_floor_price/__openerp__.py:23:10: E203 whitespace before ':'

./sale_floor_price/__openerp__.py:27:11: E203 whitespace before ':'

./sale_floor_price/__openerp__.py:28:13: E231 missing whitespace after ','

./sale_floor_price/__openerp__.py:28:23: E231 missing whitespace after ','

./sale_floor_price/floor_sale.py:69:80: E501 line too long (82 > 79 characters)
```

And the module seems to depend on a missing module: `sale_line_watcher`

File: sale_line_watcher/sale_watcher.py [hunk 1]
```diff
@@ -19,26 +19,20 @@
...
+
 
 class SaleOrderLine(Model):
     _inherit = 'sale.order.line'
 
-    def onchange_price_unit(self, cr, uid, ids,
-                            price_unit, product_id, discount, product_uom, pricelist,
-                            **kwargs):
-        '''
+    def onchange_price_unit(self, cr, uid, ids, context=None):
```
Comment (pedrobaeza) [inline sale_line_watcher/sale_watcher.py]: Why it doesn't contain price_unit parameter?

File: sale_line_watcher/sale_watcher.py [hunk 2]
```diff
@@ -19,26 +19,20 @@
...
-        res = {}
-        return res
+        """
+        return {}
 
-    def onchange_discount(self, cr, uid, ids,
-                          price_unit, product_id, discount, product_uom, pricelist,
-                          **kwargs):
-        '''
+    def onchange_discount(self, cr, uid, ids, context=None):
```
Comment (pedrobaeza) [inline sale_line_watcher/sale_watcher.py]: Why it doesn't contain discount parameter?

File: sale_line_watcher/sale_watcher.py (hunk 1 above)
Comment (pedrobaeza) [inline sale_line_watcher/sale_watcher.py]: Forget it. I see that you're using web_context_tunnel.

File: sale_line_watcher/sale_watcher.py (hunk 2 above)
Comment (pedrobaeza) [inline sale_line_watcher/sale_watcher.py]: Idem.

File: sale_line_watcher/sale_watcher.py (hunk 1 above)
Comment (pedrobaeza) [inline sale_line_watcher/sale_watcher.py]: Anyways, I see it can be useful to have an **kwargs argument to catch other possible arguments.

**Comment by mdietrichc2c** [comment]
@pedrobaeza thanks for the feedback. I changed it accordingly in sale_line_watcher, and the calls in sale_floor_price.

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by lepistone** [comment]
thanks :+1:

**Pull request merged**